python/selfie-lib/selfie_lib/ArrayMap.py

```diff
@@ -88,34 +114,40 @@ def __iter__(self) -> Iterator[K]:
     def __len__(self) -> int:
         return len(self.__data) // 2
 
-    def __binary_search_key(self, key: K) -> int:
-        # TODO: special sort order for strings
-        low, high = 0, (len(self.__data) // 2) - 1
+    def _binary_search_key(self, key: K) -> int:
+        def compare(a, b):
+            """Comparator that puts '/' first in strings."""
+            if isinstance(a, str) and isinstance(b, str):
+                a, b = a.replace("/", "\0"), b.replace("/", "\0")
+            return (a > b) - (a < b)
+
+        low, high = 0, len(self.__data) // 2 - 1
         while low <= high:
             mid = (low + high) // 2
             mid_key = self.__data[2 * mid]
-            if mid_key < key:
+            comparison = compare(mid_key, key)
+            if comparison < 0:
                 low = mid + 1
-            elif mid_key > key:
+            elif comparison > 0:
                 high = mid - 1
             else:
-                return mid
-        return -(low + 1)
+                return mid  # key found
+        return -(low + 1)  # key not found
 
```

Comment:
You have re-implemented `ListBackedSet._binarySearch` in 2 different places, and each place handles strings differently. In the Kotlin code, we are able to avoid this duplication like this:

https://github.com/diffplug/selfie-python-wip/blob/84aaf37ef84da745eceeec5ec58c8ea8c64296b8/jvm/selfie-lib/src/commonMain/kotlin/com/diffplug/selfie/ArrayMap.kt#L155-L169

The advantage of that is there is only one implementaiton of binarySearch, and it always handles Strings in a special way. For map keys and for sets.

That's what I'm trying to get to - where every comparison goes through this one function, and we can fix the string issue in just one place.